src/plugins/vis_types/vega/public/vega_view/utils.ts

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+import { ensureNoUnsafeProperties } from '@kbn/std';
+
+export function normalizeDate(date: unknown) {
+  if (typeof date === 'number') {
+    return !isNaN(date) ? date : null;
+  } else if (date instanceof Date) {
+    return date;
+  } else {
+    return normalizeObject(date);
+  }
+}
+
+/*
+Recursive function to check a nested object for a function property
+This function should run after JSON.stringify to ensure that functions such as toJSON
```

Comment:
__nit:__ :grimacing: 
```suggestion
This function should run before JSON.stringify to ensure that functions such as toJSON
```